staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/namer.go

```diff
@@ -37,6 +37,11 @@ func (n localNamer) Name(t *types.Type) string {
 	if len(n.localPackage.Package) != 0 && n.localPackage.Package == t.Name.Package {
 		return t.Name.Name
 	}
+	// For non-local and non-fundamental types, use an absolute reference
+	// see https://protobuf.com/docs/language-spec#type-references
+	if strings.Contains(t.Name.Package, ".") {
```

Comment:
Why only if the Package is more than one token (has a dot)?  Do we want to retain something like "foo.Bar", or is "foo" expected to be a fully resolved package name at this point?  I suspect it is?

I'm not super familiar with the prot generator, but I know it does unusual things with `t.Name`.

Reply:
leaving in @thockin's queue in the "changes requested" state until this is answered

otherwise, I'd be ok with tagging this for 1.31 once master reopens for merges

Reply:
It's been awhile since I wrote this but the comment is the key:

> 	// For non-local and non-fundamental types, use an absolute reference

So IIRC, I found that `t.Name.Package` could sometimes be non-packages: actual types or actually desired local references. So selecting for more than one token is a bit of a heuristic to select actual package names that are partial references.

Reply:
yah, OK.  I think I get it.